gitlab-status/handler.go

```diff
@@ -112,3 +135,88 @@ func validateRequest(req []byte) (err error) {
 
 	return nil
 }
+
+func formatLogsURL(gatewayURL string, event *sdk.Event) (string, error) {
+	systemURL, formatErr := formatSystemURL(gatewayURL)
+	if formatErr != nil {
+		return "", fmt.Errorf("error while formatting logs URL: %s", formatErr.Error())
+	}
+
+	return fmt.Sprintf("%s/dashboard/%s/%s/log?repoPath=%s/%s&commitSHA=%s",
+		systemURL, event.Owner, event.Service, event.Owner, event.Repository, event.SHA), nil
+}
+
+func formatEndpointURL(gatewayURL string, event *sdk.Event) (string, error) {
+	systemURL, formatErr := formatSystemURL(gatewayURL)
+	if formatErr != nil {
+		return "", fmt.Errorf("error while formattig endpoint URL: %s", formatErr.Error())
+	}
+	personalURL := strings.Replace(systemURL, SystemSubDomain, event.Owner, -1)
+
+	return fmt.Sprintf("%s/%s", personalURL, event.Service), nil
+}
+
+func formatDashboardURL(gatewayURL string, event *sdk.Event) (string, error) {
+	systemURL, formatErr := formatSystemURL(gatewayURL)
+	if formatErr != nil {
+		return "", fmt.Errorf("error while formatting dashboard URL: %s", formatErr.Error())
+	}
+
+	return fmt.Sprintf("%s/dashboard/%s", systemURL, event.Owner), nil
+}
+
+func formatSystemURL(gatewayURL string) (string, error) {
+	if strings.HasSuffix(gatewayURL, "/") {
+		gatewayURL = strings.TrimSuffix(gatewayURL, "/")
+	}
+	subDomain, err := getSubDomain(gatewayURL)
+	if err != nil {
+		return "", fmt.Errorf("error while geting subdomain for system URL: %s", err)
+	}
+	systemURL := strings.Replace(gatewayURL, subDomain, SystemSubDomain, -1)
+	return systemURL, nil
+}
+
+func getSubDomain(URL string) (string, error) {
+	parsedURL, parseErr := url.Parse(URL)
+	if parseErr != nil {
+		return "", fmt.Errorf("Unable to parse URL: %s", parseErr.Error())
+	}
+	subdomain := strings.Split(parsedURL.Host, ".")
+
+	//Host is www.world.org and subdomain would be www aka. 0th element of the slice
+	return subdomain[0], nil
```

Comment:
This needs range checking. What if there are 0 elements in the slice/array? then throw an error.

Reply:
Add a test-case for this too.

Reply:
I actually have test for the case [here](https://github.com/openfaas/openfaas-cloud/blob/fb8e73e9d3a13493adbda3f224e35aed08debb9c/gitlab-status/handler_test.go#L324)

Reply:
What will happen if `gateway_public_url` is configured without subdomain and looks like `some.path.with-dots`?
Is this possible use-case?
If yes, we would like only to prefix the url, instead of replacing.

Reply:
I think and hope the structure of subdmain.o5s.io that way is a convention already, also we use the same convention when we have pretty_url for every user, I think people would like to have full cloud functionality when deploying it and setting 3 items separated with dots is the way, additional is that we name the functions from the cloud which are exposed in the repo has naming system-gitlab-event instead of gitlab-event so the router can make use of the subdomain system. And I think I saw somewhere auth.system as variable replacing system in the public URL.

Reply:
This ^ is just to show the comment as answer to the question above

Reply:
@ivanayov are you ok with this?